addons/io_scene_gltf2/__init__.py

```diff
@@ -69,6 +69,17 @@ def reload_package_recursive(current_dir, module_dict):
 
 extension_panel_unregister_functors = []
 
+
+def on_export_format_changed(self, context):
+    # Update the file extension when the format (.glb/.gltf) changes
+    sfile = context.space_data
+    operator = sfile.active_operator
+    if operator.check(context):
+        # Weird hack to force the filepicker to notice filename changed
+        bpy.ops.file.filenum(increment=1)
+        bpy.ops.file.filenum(increment=-1)
```

Comment:
Check out this hack. Took forever to come up with.

Reply:
It does mean trailing 0s get erased when switching formats though. That sucks :/